test/Microsoft.Azure.WebJobs.Host.EndToEndTests/Scale/ScaleHostEndToEndTests.cs

```diff
@@ -86,28 +93,38 @@ public async Task ScaleManager_GetScaleStatusAsync_ReturnsExpected(bool tbsEnabl
                 services.AddSingleton<IConcurrencyStatusRepository, TestConcurrencyStatusRepository>();
             })
             .ConfigureWebJobsScale((context, builder) =>
-                {
-                    builder.UseHostId(hostId);
-                },
-                scaleOptions =>
-                {
-                    scaleOptions.IsTargetScalingEnabled = tbsEnabled;
-                    scaleOptions.MetricsPurgeEnabled = false;
-                    scaleOptions.ScaleMetricsMaxAge = TimeSpan.FromMinutes(4);
-                    scaleOptions.IsRuntimeScalingEnabled = true;
-                });
-
-            foreach (TriggerMetadata t in triggerMetadata)
             {
-                if (t.Type == ATriggerType)
-                {
-                    hostBuilder.ConfigureTestExtensionAScale(t);
-                }
-                else
+                builder.UseHostId(hostId);
+
+                // Get the interface type you want to find
+                Type interfaceType = typeof(IWebJobsStartup2);
+
+                // Get all the loaded assemblies in the current app domain
+                Assembly[] assemblies = AppDomain.CurrentDomain.GetAssemblies();
+
+                foreach (Assembly assembly in assemblies)
                 {
-                    hostBuilder.ConfigureTestExtensionBScale(t);
+                    var attributes = assembly.GetCustomAttributes<WebJobsStartupAttribute>();
+
+                    foreach (var attribute in attributes)
+                    {
+                        if (interfaceType.IsAssignableFrom(attribute.WebJobsStartupType))
+                        {
+                            // Filter out triggers needed to be be configured
+                            var triggersToConfigure = triggerMetadata.Where(x => x.Type == attribute.Name);
```

Comment:
If we accept the default Name from existing attribute definitions, are we sure these lookups will always match? I.e. we need to verify that the source of the TriggerMetadata.Type is the same as we're using here.

Reply:
I add ScaleHostBuilderContext.GetTriggersMetadata(string type) instead.